### PR TITLE
add median of matrix

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by BamaCharanChhandogi** [comment]
This is the first PR of the median of the matrix.
@cclauss could you check, please?

**Comment by BamaCharanChhandogi** [comment]
@cclauss please check

**Comment by BamaCharanChhandogi** [comment]
@cclauss check, please!

File: matrix/median_matrix.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        >>> matrix = [[1, 2, 3], [4, 5, 6]]
+        >>> median(matrix)
+        3
+    """
+    # Flatten the matrix into a 1D list
+    linear = [num for row in matrix for num in row]
+
+    # Sort the 1D list
+    linear.sort()
```
Comment (cclauss) [inline matrix/median_matrix.py:29]:
```suggestion
    # Flatten the matrix into a sorted 1D list
    linear = sorted(num for row in matrix for num in row)
```

File: matrix/median_matrix.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    # Calculate the middle index
+    mid = (0 + len(linear) - 1) // 2
```
Comment (cclauss) [inline matrix/median_matrix.py:32]:
```suggestion
    mid = (len(linear) - 1) // 2
```

**Review by cclauss — APPROVED** [review]
Let's merge this!!!

**Pull request merged**